Etiquette/CODE_REVIEW.md

```diff
@@ -0,0 +1,108 @@
+* Keep pull requests small, ideally it shouldn't be more than **800 additions/deletions**
+* Create a draft PR for work that is in progress, but only do that if you seek for opinions on your work
+* Mark PR with "Blocked" label if it has any dependency that should stop it from being merged (it's fine for pull request be both "blocked" and "ready for review", but it shouldn't be "ready for review" and "wip")
+* Give your team mates a reasonable time to review your changes, if the change is critical use "Top priority" label and post it in the team channel
+* Do not leave stale pull requests, seek for help/actions or close them (make sure you don't delete your local branch if you may need it later)
+* It's our common effort to push product forward and we all do equally valuable work. That in mind when creating PR try to review some PRs, avoid having more than X opened PRs
+
+
+Reviewing Pull Request
+--------
+
+* Accept that many programming decisions are opinions. Discuss tradeoffs, which
+  you prefer, and reach a resolution quickly. There are always multiple ways of doing things, and they may be equally acceptable
+* Focus not on finding flaws in the code (no one writes perfect code) but on understanding the change in the first place
+* Encourage things you like, try not only comment on what you don't like
+* Before suggesting to change code think if it brings any real value or if it is just your personal preference. Is this code something you can live with or not?
+* Do not try to make everything perfect (it never will be and otherwise we'll never deliver anything), but do not compromise on quality
+* Do not comment if you don't have any suggestions to address your comment
+* Use github suggestions as much as possible
+* If you feel strong against the change - request changes and explain why you are doing that. Do not request changes for typos, missing documentation comments or code style violations, trust your colleagues to address them.
+* Approve changes if you are confident
+* Comment on the changes if something is not clear to you and needs further clarification or actions
+* If you are not familiar with the domain ask for more context (better in person) rather than skipping to something more familiar
+* Ask, don't tell. ("What do you think about...?", "How about ...?", "Should we try ...?")
+* Be breif but clear in your comments (code review is not about proving who is more correct or knows more)
+* Unless explicitly asked for it do not suggest code to be written in a specific way - let the author to find or suggest a solution themselves, they may come up with a better one
+* Submit all the comments as a single review, even if there is just one comment
+* If you see a stale pull request without any updates - ask for the status update in the comment
+* Do not be a blocker - if you requested changes and they were addressed make sure you review it
+* Do not merge others pull requests - they may be waiting for some one else review
```

Comment:
```suggestion
* Do not merge others pull requests - they may be waiting for someone else review
```

[ ❔] Given we use a merge bot, should this rule be re-worded accordingly? Also not sure this rule apply to us, as to my understanding our own etiquette says that we should not merge PRs ourselves, and add the "Merge" label when we are the 2nd approver of the PR and there is no pending discussion or doubt? (which is rightfully mentioned in "Responding to feedback" section below already btw 👍 )

Reply:
> My personal preference is that only the authors can merge

I agree with that (and so should change the line about adding "merge" label), but again it feels like there is no one common approach here, so having this written will help in settling that

Reply:
@AliSoftware the "Responding to feedback" section which is from the point of author of the changes, so that means that author should be the one adding this label (there always can be exception like if they are on holidays but change is good to go otherwise)

Reply:
Ok. I still think that the implicit "Only merge your own PRs" here (as opposed to "don't merge other's PRs") might be confusing to new joiners (they might think of it actually meaning "hitting the Merge button on GitHub")

```suggestion
* Do not merge others pull requests - they may be waiting for someone else review. In general, the PR author should be the one adding the "Merge" label to merge their own PRs (except rare cases like if the author is on holidays but the PR is otherwise good to go)
```